lib/avrogen/schema/schema_registry.ex

```diff
@@ -39,10 +40,27 @@ defmodule Avrogen.Schema.SchemaRegistry do
     end)
     |> Avrogen.Schema.topological_sort()
     |> Noether.Either.map(fn schemas ->
-      json = Jason.encode!(schemas)
-      encoder = make_encoder(json)
-      decoder = make_decoder(json)
-      :ets.insert(__MODULE__, {@ets_name, json, encoder, decoder})
+      try do
+        json = Jason.encode!(schemas)
+        encoder = make_encoder(json)
+        decoder = make_decoder(json)
+        :ets.insert(__MODULE__, {@ets_name, json, encoder, decoder})
+      rescue
+        e ->
+          formatted_error = Exception.format(:error, e, __STACKTRACE__)
+          json = Jason.encode(schemas) |> elem(1)
+
+          Logger.error(
+            "Error when attempting to make encoder/decoder: #{formatted_error}; schemas: #{inspect(schemas)} schemas_json: #{json}",
```

Comment:
`schemas` is very long so might get truncated by `inspect`.

`IO.inspect(schemas, limit: :infinity, printable_limit: :infinity)`